Gem/Code/Source/Components/NetworkWeaponsComponent.cpp

```diff
@@ -360,11 +363,11 @@ namespace MultiplayerSample
                 const AZ::Vector3& aimAngles = GetNetworkSimplePlayerCameraComponentController()->GetAimAngles();
                 const AZ::Quaternion aimRotation =
                     AZ::Quaternion::CreateRotationZ(aimAngles.GetZ()) * AZ::Quaternion::CreateRotationX(aimAngles.GetX());
-                // TODO: This should probably be a physx raycast out to some maxDistance
                 const AZ::Vector3 fwd = AZ::Vector3::CreateAxisY();
-                const AZ::Vector3 aimTarget = worldTm.GetTranslation() + aimRotation.TransformVector(fwd * 5.0f);
-                AZ::Vector3 aimSource = weaponInput->m_shotStartPosition;
-
+                AZ::Vector3 baseCameraOffset;
+                AZ::Interface<AZ::IConsole>::Get()->GetCvarValue("cl_cameraOffset", baseCameraOffset);
+                const AZ::Vector3 cameraOffset = aimRotation.TransformVector(baseCameraOffset);
```

Comment:
You know if you had a second cl_cameraPivotOffset that was used for the point around which the camera pivoted you could use that as the start position for the raycast - and when you support zooming the camera in/out and unoccluding it will work a lot better.

Reply:
I was thinking we could do something like this. We could take the depth of the firing bone relative to the player transform vs that of the camera offset. I still think we want the dot product check to ensure we're not doing something like firing at something directly to the right of the weapon or something.